src/unaudited/ECDSAStakeRegistry.sol

```diff
@@ -0,0 +1,482 @@
+        bytes memory _signature
+    ) internal view {
+        if (!_signer.isValidSignatureNow(_dataHash, _signature)) {
+            revert InvalidSignature();
+        }
+    }
+
+
+    /// @notice Retrieves the operator weight for a signer, either at the last checkpoint or a specified block.
+    /// @param _signer The address of the signer whose weight is returned.
+    /// @param _referenceBlock The block number to query the operator's weight at, or the maximum uint32 value for the last checkpoint.
+    /// @return The weight of the operator.
+    function _getOperatorWeight(
+        address _signer,
+        uint32 _referenceBlock
+    ) internal view returns (uint256) {
+        if (_referenceBlock == type(uint32).max) {
+            return _operatorWeightHistory[_signer].latest();
+        } else {
+            return _operatorWeightHistory[_signer].getAtBlock(_referenceBlock);
+        }
+    }
+
+    /// @notice Retrieve the total stake weight at a specific block or the latest if not specified.
+    /// @dev If the `_referenceBlock` is the maximum value for uint32, the latest total weight is returned.
+    /// @param _referenceBlock The block number to retrieve the total stake weight from.
+    /// @return The total stake weight at the given block or the latest if the given block is the max uint32 value.
+    function _getTotalWeight(uint32 _referenceBlock) internal view returns (uint256) {
+        if (_referenceBlock == type(uint32).max) {
+            return _totalWeightHistory.latest();
+        } else {
+            return _totalWeightHistory.getAtBlock(_referenceBlock);
+        }
+    }
+
+    /// @notice Retrieves the threshold stake for a given reference block.
+    /// @param _referenceBlock The block number to query the threshold stake for.
+    /// If set to the maximum uint32 value, it retrieves the latest threshold stake.
+    /// @return The threshold stake in basis points for the reference block.
+    function _getThresholdStake(uint32 _referenceBlock) internal view returns (uint256) {
+        if (_referenceBlock == type(uint32).max) {
+            return _thresholdWeightHistory.latest();
+        } else {
+            return _thresholdWeightHistory.getAtBlock(_referenceBlock);
+        }
+    }
+
+    /// @notice Validates that the cumulative stake of signed messages meets or exceeds the required threshold.
+    /// @param _signedWeight The cumulative weight of the signers that have signed the message.
+    /// @param _referenceBlock The block number to verify the stake threshold for
+    function _validateThresholdStake(uint256 _signedWeight, uint32 _referenceBlock) internal view {
+        uint256 totalWeight = _getTotalWeight(_referenceBlock);
+        if (_signedWeight > totalWeight){
+            revert InvalidSignedWeight();
+        }
+        uint256 thresholdStake = _getThresholdStake(_referenceBlock);
+        if ((thresholdStake * BPS)/ totalWeight > (_signedWeight * BPS) / totalWeight){
```

Comment:
looks like you actually totally changed the meaning of `thresholdStake` to be the literal amount required to sign?
this check could now just be `if( thresholdStake > _signedWeight )` because you doing the exact same manipulations on both sides of the inequality

Reply:
Addressed in [79d95aa](https://github.com/Layr-Labs/eigenlayer-middleware/pull/194/commits/79d95aaf40f7b27f85ced4c42d2fce2d2086a865)